sync/sync.go

```diff
@@ -53,6 +58,10 @@ func (s *Synchronizer) fetcherTask(ctx context.Context, height uint64, verifiers
 			if err != nil {
 				continue
 			}
+			highestBlock, err := s.StarknetData.BlockByID(ctx, "latest")
```

Comment:
This is a bit concerning. We would be fetching two blocks every for block. This could slow down the sync, and create a lot more network traffic that users would be paying for.

Reply:
How do you suggest we handle latest block? Probably only poll it at intervals? Or is there a way to listen to new blocks on Starknet?

Reply:
> Or is there a way to listen to new blocks on Starknet?

I dont think there is.

How about this;
Let's query the latest block when we are starting to sync and only query it again after we reach that initial latest block.
So after storing a block; if `storedBlockHeight >= highestBlock` is true, query new highest block.

Reply:
Thanks. Will do this then

Reply:
Done